### PR TITLE
Fix for fuzzer crash testcase

### PR DESCRIPTION
**Pull request by qxliu76**
Add a check for stringOffSet(uint16) overflow, return early if overflow happens

### CODE REVIEW

File: src/hb-ot-name-table.hh [hunk 1]
```diff
@@ -230,6 +228,7 @@ struct name
     TRACE_SANITIZE (this);
     return_trace (c->check_struct (this) &&
 		  likely (format == 0 || format == 1) &&
+		  likely (count <= 16382) &&
```
Comment (garretrieger) [inline src/hb-ot-name-table.hh]: Can you add a comment explaining how this number is derived and the reason for the check.

File: src/hb-ot-name-table.hh [hunk 2]
```diff
@@ -197,8 +197,6 @@ struct name
 
     if (unlikely (c->ran_out_of_room)) return_trace (false);
 
-    assert (this->stringOffset == c->length ());
```
Comment (behdad) [inline src/hb-ot-name-table.hh]: Why remove?  If overflow happens, we should detect it earlier and return.

Comment (behdad) [reply src/hb-ot-name-table.hh]: The line a few lines earlier where we have this:
```
   this->stringOffset = c->length ();
```
It should become something like:
```
  if (unlikely (!c->check_assign (this->stringOffset, c->length ())) return_trace (false);
```

File: src/hb-ot-name-table.hh [hunk 3]
```diff
@@ -230,6 +228,7 @@ struct name
     TRACE_SANITIZE (this);
     return_trace (c->check_struct (this) &&
 		  likely (format == 0 || format == 1) &&
+		  likely (count <= 16382) &&           /*make sure stringOffSet(uint16) won't overflow, 6 bytes (header) + 4 bytes (name record) * count <= 65535*/
```
Comment (behdad) [inline src/hb-ot-name-table.hh]: We shouldn't need this.

File: src/hb-ot-name-table.hh (hunk 2 above)
Comment (qxliu76) [reply src/hb-ot-name-table.hh]: oh, so you don't check in sanitize() for invalid tables but rather do checks in serializer.

Comment (qxliu76) [reply src/hb-ot-name-table.hh]: sorry, I have a question here: 
we have a task to remove return type of bool from all serialize calls. Is serializer be able to capture this error state in this case?

Comment (behdad) [reply src/hb-ot-name-table.hh]: Yes, the `check_assign` thing records the failure.

Comment (behdad) [reply src/hb-ot-name-table.hh]: > oh, so you don't check in sanitize() for invalid tables but rather do checks in serializer.

The job of `sanitize` is only to make sure the rest of the methods to *access* the table don't cause invalid memory access.  We are not in the font law enforcement business :D.

Comment (qxliu76) [reply src/hb-ot-name-table.hh]: got it. Thanks! :)

**Pull request merged**